### ISSUE TITLE
`coupledIteration` stored on database is different than that used during the run

### ISSUE THREAD

**Issue by keckler** [terrapower/armi#1962]
In the code chunk below, we store `coupledIteration` as `n+1`, whereas we send `n` into `interactAllCOupled()`:
https://github.com/terrapower/armi/blob/eef58ddcd797c7acc542a4941b95dcb84dda2670/armi/operators/operator.py#L456-L458

This also means that the iteration number that is printed in the STDOUT is not aligned with the value used during `interactCoupled()` calls. In the screenshot below, the circled value comes from the DB.
![image](https://github.com/user-attachments/assets/ee004278-d008-461f-be28-b5a2bc270e65)
https://github.com/terrapower/armi/blob/eef58ddcd797c7acc542a4941b95dcb84dda2670/armi/operators/operator.py#L568-L573

So basically, if anyone is trying to do something within an `interactCoupled()` call that is dependent on the iteration number, they are very probably using the wrong iteration number.

**Comment by keckler** [comment]
The `LatticePhysicsInterface` also assumes zero-indexing:
https://github.com/terrapower/armi/blob/eef58ddcd797c7acc542a4941b95dcb84dda2670/armi/physics/neutronics/latticePhysics/latticePhysicsInterface.py#L224-L257

**Comment by keckler** [comment]
I'm not sure that anything important is actually broken here, so I removed the "bug" label. But I'd say this is confusing and I would advocate that the value used and the value printed are aligned with each other.

**Comment by keckler** [comment]
The "Tight Coupling Convergence Summary" table is also zero-indexed:
![image](https://github.com/user-attachments/assets/3462cfba-bfe8-4d29-9ca3-c33914f95cab)

So it really seems like everything except for the interaction hook header is zero-indexed. Seems like an easy fix to just change the headers.
